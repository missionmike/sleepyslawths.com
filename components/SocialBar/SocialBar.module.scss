@import 'styles/mixins.scss';

.socialBar {
  margin: 2rem 0;
  display: flex;
  flex-direction: row;
  gap: 0.8rem;

  @include media-breakpoint-down(sm) {
    justify-content: space-between;
  }
}
