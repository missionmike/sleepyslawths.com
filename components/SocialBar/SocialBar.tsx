import { IconFacebook } from './icons/IconFacebook';
import { IconInstagram } from './icons/IconInstagram';
import { IconTikTok } from './icons/IconTikTok';
import { IconX } from './icons/IconX';
import { IconYouTube } from './icons/IconYouTube';
import React from 'react';
import { siteMetadata } from 'data/siteMetadata';
import styles from './SocialBar.module.scss';

/**
 * Social media icons component to link to various
 * platforms.
 *
 * @returns {React.ReactElement} Social media icons with links.
 */
const SocialBar = () => {
  return (
    <div className={styles.socialBar}>
      <IconYouTube url={siteMetadata.youTube} />
      <IconTikTok url={siteMetadata.tikTok} />
      <IconInstagram url={siteMetadata.instagram} />
      <IconFacebook url={siteMetadata.facebook} />
      <IconX url={siteMetadata.twitter} />
    </div>
  );
};

export { SocialBar };
