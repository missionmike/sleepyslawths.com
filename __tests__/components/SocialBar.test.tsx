import React from 'react';
import { SocialBar } from 'components/SocialBar/SocialBar';
import { render } from '@testing-library/react';

describe('SocialBar component', () => {
  it('renders the component', () => {
    const { container } = render(<SocialBar />);
    expect(container).toMatchSnapshot();
  });
});
