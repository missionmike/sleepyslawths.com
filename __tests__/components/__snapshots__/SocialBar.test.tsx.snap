// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialBar component renders the component 1`] = `
<div>
  <div
    class="socialBar"
  >
    <a
      class="icon"
      href="https://www.youtube.com/@sleepyslawths"
      rel="noopener noreferrer"
      target="_blank"
      title="Subscribe to Sleepy Slawths on YouTube"
    >
      <svg
        class="svg"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.3,13.5H8.3v5H7.4v-5H6.5v-0.9h2.8L9.3,13.5L9.3,13.5z M14.7,15.2v2.4c0,0.5-0.2,1-0.7,1 c-0.3,0-0.5-0.1-0.8-0.4v0.3h-0.8v-5.9h0.8v1.9c0.2-0.2,0.4-0.4,0.7-0.4C14.5,14.2,14.7,14.7,14.7,15.2z M13.8,15.3 c0-0.1,0-0.3-0.1-0.3c-0.1-0.1-0.3-0.1-0.4-0.1L13.2,15v2.7l0.2,0.2c0.1,0.1,0.3,0.1,0.4,0c0-0.1,0.1-0.2,0.1-0.3L13.8,15.3 L13.8,15.3z M11.5,9.4c0.2,0,0.4-0.2,0.4-0.4V6.9c0-0.3-0.1-0.4-0.4-0.4c-0.2,0-0.4,0.2-0.4,0.4V9C11.1,9.2,11.3,9.4,11.5,9.4z M10.8,17.5c-0.1,0.1-0.3,0.3-0.5,0.3c-0.2,0-0.2-0.1-0.2-0.3v-3.3H9.3v3.6c0,0.9,0.6,0.9,1,0.6c0.2-0.1,0.3-0.2,0.5-0.4v0.5h0.8 v-4.3h-0.8C10.8,14.2,10.8,17.5,10.8,17.5z M24,0v19c0,2.8,0,5,0,5H5c-2.8,0-5,0-5,0V5c0-2.8,0-5,0-5h19C21.8,0,24,0,24,0z M13.4,9.4c0,0.4,0.1,0.8,0.6,0.8c0.3,0,0.6-0.1,1-0.6v0.5h0.8V5.6H15v3.5c-0.1,0.1-0.3,0.3-0.5,0.3c-0.2,0-0.2-0.1-0.2-0.3V5.6 h-0.8C13.4,5.6,13.4,9.4,13.4,9.4z M10.3,6.8V9c0,0.8,0.4,1.2,1.2,1.2c0.7,0,1.2-0.5,1.2-1.2V6.8c0-0.7-0.5-1.2-1.2-1.2 C10.8,5.6,10.3,6.1,10.3,6.8z M7.1,4l1.1,3.7v2.5h1V7.7L10.3,4h-1L8.7,6.5L8.1,4C8.1,4,7.1,4,7.1,4z M18.7,15.7c0-3-0.2-4.1-2.1-4.2 c-1.7-0.1-7.4-0.1-9.1,0c-1.9,0.1-2.1,1.3-2.1,4.2c0,3,0.2,4.1,2.1,4.2c1.7,0.1,7.4,0.1,9.1,0C18.4,19.8,18.7,18.6,18.7,15.7z M17.7,15.4v1.1h-1.5v0.8c0,0.3,0,0.6,0.4,0.6c0.3,0,0.4-0.2,0.4-0.6V17h0.8v0.3c0,0.8-0.4,1.3-1.2,1.3c-0.8,0-1.2-0.6-1.2-1.3v-1.9 c0-0.8,0.5-1.3,1.2-1.3C17.3,14.1,17.7,14.6,17.7,15.4z M16.8,15.4c0-0.3-0.1-0.5-0.4-0.5c-0.3,0-0.4,0.2-0.4,0.5v0.4h0.7V15.4z"
          fill="#ff0000"
        />
      </svg>
       
    </a>
    <a
      class="icon"
      href="https://www.tiktok.com/@sleepyslawths"
      rel="noopener noreferrer"
      target="_blank"
      title="Follow Mission Mike on TikTok"
    >
      <svg
        clip-rule="evenodd"
        fill-rule="evenodd"
        image-rendering="optimizeQuality"
        shape-rendering="geometricPrecision"
        text-rendering="geometricPrecision"
        viewBox="0 0 3333 3333"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3021 0H313C141 0 0 141 0 312v2708c0 172 141 312 313 312h2708c172 0 313-141 313-312V312c0-172-141-312-313-312zm-946 623c35 296 200 473 487 492v333c-167 16-313-38-483-141v623c0 792-863 1040-1210 472-223-365-87-1006 629-1032v352c-54 9-113 23-166 40-159 54-249 155-224 332 48 341 674 441 621-225V622h345z"
        />
      </svg>
    </a>
    <a
      class="icon"
      href="https://www.instagram.com/sleepyslawths"
      rel="noopener noreferrer"
      target="_blank"
      title="Follow Mission Mike on Instagram"
    >
      <div
        style="width: 0px; height: 0px;"
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <radialgradient
              cx="162.4"
              cy="-448.2"
              gradientTransform="matrix(0 -1.982 -1.8439 0 -791.38 463.95)"
              gradientUnits="userSpaceOnUse"
              id="ig-svg-d"
              r="66"
            >
              <stop
                class="ig-svg-a"
                offset="0"
              />
              <stop
                class="ig-svg-a"
                offset=".1"
              />
              <stop
                class="ig-svg-b"
                offset=".5"
              />
              <stop
                class="ig-svg-c"
                offset="1"
              />
            </radialgradient>
            <radialgradient
              cx="167.5"
              cy="-456"
              gradientTransform="matrix(0 -1.982 -1.8439 0 -791.38 463.95)"
              gradientUnits="userSpaceOnUse"
              id="ig-svg-c"
              r="9"
            >
              <stop
                class="ig-svg-a"
                offset="0"
              />
              <stop
                class="ig-svg-a"
                offset=".1"
              />
              <stop
                class="ig-svg-b"
                offset=".5"
              />
              <stop
                class="ig-svg-c"
                offset="1"
              />
            </radialgradient>
            <radialgradient
              cx="234"
              cy="-453.3"
              gradientTransform="matrix(0 -1.982 -1.8439 0 -791.38 463.95)"
              gradientUnits="userSpaceOnUse"
              id="ig-svg-b"
              r="7.5"
            >
              <stop
                class="ig-svg-a"
                offset="0"
              />
              <stop
                class="ig-svg-a"
                offset=".1"
              />
              <stop
                class="ig-svg-b"
                offset=".5"
              />
              <stop
                class="ig-svg-c"
                offset="1"
              />
            </radialgradient>
            <radialgradient
              cx="139.5"
              cy="-347.9"
              gradientTransform="matrix(.1739 .8687 3.5818 -.7172 1193.4 -365.3)"
              gradientUnits="userSpaceOnUse"
              id="ig-svg-a2"
              r="70.9"
            >
              <stop
                class="ig-svg-d"
                offset="0"
              />
              <stop
                class="ig-svg-d"
                offset=".128"
              />
              <stop
                offset="1"
                stop-color="#60F"
                stop-opacity="0"
              />
            </radialgradient>
          </defs>
        </svg>
      </div>
      <svg
        class="svg"
        viewBox="0 0 132 132"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="ig-svg-st0"
          d="m131.9 131.9c0.1-1.1 0.1-47.8 0.1-65.9s0-64.8-0.1-66h-131.9v66c0 27.5 0 64.5 0.2 66l131.7-0.1z"
        />
        <path
          class="ig-svg-st1"
          d="m30.9 131.9c1.6 0.1 18.3 0.1 35 0.1s33.3 0 34.9-0.1"
        />
        <path
          class="ig-svg-st2"
          d="M87.5 0H66C38.5 0 30.4 0 28.8 0.2"
        />
        <path
          class="ig-svg-st3"
          d="m131.9 103.3c0.1-1.1 0.1-19.3 0.1-37.4s0-64.8-0.1-66h-65.9c-27.5 0.1-64.4 0.1-66 0.3v44.4 21.4c0 27.5 0 35.6 0.2 37.2v28.7c1.6 0.1 49.1 0.1 65.7 0.1s64.5 0 66.1-0.1l-0.1-28.6z"
        />
        <path
          class="ig-svg-st4"
          d="m66 17.3c-13.2 0-14.9 0.1-20.1 0.3s-8.7 1.1-11.8 2.3c-3.2 1.2-5.9 2.9-8.6 5.6s-4.4 5.4-5.6 8.6c-1.2 3.1-2 6.6-2.3 11.8-0.2 5.2-0.3 6.9-0.3 20.1s0.1 14.9 0.3 20.1 1.1 8.7 2.3 11.8c1.2 3.2 2.9 5.9 5.6 8.6s5.4 4.4 8.6 5.6c3.1 1.2 6.6 2 11.8 2.3 5.2 0.2 6.9 0.3 20.1 0.3s14.9-0.1 20.1-0.3 8.7-1.1 11.8-2.3c3.2-1.2 5.9-2.9 8.6-5.6s4.4-5.4 5.6-8.6c1.2-3.1 2-6.6 2.3-11.8 0.2-5.2 0.3-6.9 0.3-20.1s-0.1-14.9-0.3-20.1-1.1-8.7-2.3-11.8c-1.2-3.2-2.9-5.9-5.6-8.6s-5.4-4.4-8.6-5.6c-3.1-1.2-6.7-2-11.8-2.3-5.2-0.3-6.9-0.3-20.1-0.3zm-4.4 8.7h4.4c13 0 14.6 0 19.7 0.3 4.8 0.2 7.3 1 9 1.7 2.3 0.9 3.9 1.9 5.6 3.6s2.8 3.3 3.6 5.6c0.7 1.7 1.5 4.3 1.7 9 0.2 5.1 0.3 6.7 0.3 19.7s-0.1 14.6-0.3 19.7c-0.2 4.8-1 7.3-1.7 9-0.9 2.3-1.9 3.9-3.6 5.6s-3.3 2.8-5.6 3.6c-1.7 0.7-4.3 1.5-9 1.7-5.1 0.2-6.7 0.3-19.7 0.3s-14.6-0.1-19.7-0.3c-4.8-0.2-7.3-1-9.1-1.7-2.3-0.9-3.9-1.9-5.6-3.6s-2.8-3.3-3.6-5.6c-0.7-1.7-1.5-4.3-1.7-9-0.2-5.1-0.3-6.6-0.3-19.6s0-14.6 0.3-19.7c0.2-4.8 1-7.3 1.7-9 0.9-2.3 1.9-3.9 3.6-5.6s3.3-2.8 5.6-3.6c1.7-0.7 4.3-1.5 9.1-1.7 4.5-0.3 6.2-0.4 15.3-0.4zm30.4 8.1c-3.2 0-5.8 2.6-5.8 5.8s2.6 5.8 5.8 5.8 5.8-2.6 5.8-5.8-2.6-5.8-5.8-5.8zm-26 6.9c-13.8 0-25 11.2-25 25s11.2 25 25 25 25-11.2 25-25-11.2-25-25-25zm0 8.8c9 0 16.2 7.3 16.2 16.2 0 9-7.3 16.2-16.2 16.2-9 0-16.2-7.3-16.2-16.2 0-9 7.2-16.2 16.2-16.2z"
        />
      </svg>
       
    </a>
    <a
      class="icon"
      href="https://facebook.com/sleepyslawths"
      rel="noopener noreferrer"
      target="_blank"
      title="Follow Mission Mike on Facebook"
    >
      <svg
        class="svg"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22.7,0H0c0,0,0,0.6,0,1.3v21.4C0,23.4,0,24,0,24h12.8v-9.3H9.7v-3.6h3.1V8.4c0-3.1,1.9-4.8,4.7-4.8 c1.3,0,2.5,0.1,2.8,0.1V7l-1.9,0c-1.5,0-1.8,0.7-1.8,1.8v2.3h3.6l-0.5,3.6h-3.1V24h6.1c0.7,0,1.3,0,1.3,0V0C24,0,23.4,0,22.7,0z"
          fill="#3B5998"
        />
      </svg>
       
    </a>
    <a
      class="icon"
      href="https://twitter.com/sleepyslawths"
      rel="noopener noreferrer"
      target="_blank"
      title="Follow Sleepy Slawths on X!"
    >
      <img
        alt="X icon"
        class="iconX"
        data-nimg="1"
        decoding="async"
        height="30"
        loading="lazy"
        src="/_next/image?url=%2Fstatic%2Fimages%2Ficons%2Fx.png&w=64&q=75"
        srcset="/_next/image?url=%2Fstatic%2Fimages%2Ficons%2Fx.png&w=32&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Ficons%2Fx.png&w=64&q=75 2x"
        style="color: transparent;"
        width="30"
      />
    </a>
  </div>
</div>
`;
